javalib/src/main/scala/java/net/WindowsNet.scala

```diff
@@ -0,0 +1,309 @@
+      interf: NetworkInterface,
+      source: InetAddress
+  ): Unit = {
+    require(group.isMulticastAddress(), "Address is not a multicast address")
+    if (source != null) {
+      require(
+        !source.isAnyLocalAddress(),
+        "Source address is a wildcard address"
+      )
+      require(
+        !source.isMulticastAddress(),
+        "Source address is multicast address"
+      )
+      require(
+        source.getClass == group.getClass,
+        "Source address is different type to group"
+      )
+    }
+
+    val (name, optName) = membership match {
+      case Net.Membership.Join => ("IP_ADD_MEMBERSHIP", ip.IP_ADD_MEMBERSHIP)
+      case Net.Membership.Drop => ("IP_DROP_MEMBERSHIP", ip.IP_DROP_MEMBERSHIP)
+    }
+
+    val (optValue, optLen) = (family, group) match {
+      case (StandardProtocolFamily.INET, group: Inet4Address) =>
+        val mName = stackalloc[ip.ip_mreq]()
+        if (interf != null) {
+          val ifAddrs = interf.getInetAddresses()
+          if (!ifAddrs.hasMoreElements()) {
+            throw new SocketException(
+              s"bad argument for $name: No IP addresses bound to interface"
+            )
+          }
+          val addrPtr = group.getAddress().at(0).asInstanceOf[Ptr[in.in_addr_t]]
+          val ifAddrPtr =
+            ifAddrs
+              .nextElement()
+              .getAddress()
+              .at(0)
+              .asInstanceOf[Ptr[in.in_addr_t]]
+          mName.imr_multiaddr.s_addr = inet.htonl(!addrPtr)
+          mName.imr_address.s_addr = inet.htonl(!ifAddrPtr)
+        } else {
+          val addrPtr = group.getAddress().at(0).asInstanceOf[Ptr[in.in_addr_t]]
+          val ifAddrPtr = getMulticastInterfaceOption(fd).getAddress
+            .getAddress()
+            .at(0)
+            .asInstanceOf[Ptr[in.in_addr_t]]
+          mName.imr_multiaddr.s_addr = inet.htonl(!addrPtr)
+          mName.imr_address.s_addr = inet.htonl(!ifAddrPtr)
+        }
+        (mName.asInstanceOf[Ptr[Byte]], sizeof[ip.ip_mreq].toUInt)
+      // case (StandardProtocolFamily.INET6, group: Inet6Address) =>
+      case _ =>
+        throw new IllegalArgumentException("Address type not supported")
+    }
+
+    if (posix.sys.socket.setsockopt(
+          fd.fd,
+          in.IPPROTO_IP,
+          optName,
+          optValue,
+          optLen
+        ) != 0) {
+      throw new SocketException(
+        s"Exception while setting socket option $name, errno: ${WSAGetLastError()}"
+      )
+    }
+  }
+
+  override def close(fd: FileDescriptor): Unit = closeSocket(fd.handle)
+
+  override def localAddress(
+      fd: FileDescriptor,
+      family: ProtocolFamily
+  ): InetSocketAddress = {
+    val len = stackalloc[unixsocket.socklen_t]()
+    val saddr = family match {
+      case StandardProtocolFamily.INET =>
+        !len = sizeof[in.sockaddr_in].toUInt
+        stackalloc[in.sockaddr_in]().asInstanceOf[Ptr[unixsocket.sockaddr]]
+//      case StandardProtocolFamily.INET6 =>
```

Comment:
Might want to leave a comment as to why code is commented out.
Currently leaves the reader wondering why.  

It looks like that code should work on Windows. Is there a 
Windows specific difference?

If it is commented out in the current mainline SN code, that may
have been my paranoia about leaving un-exercised code enabled.

I believe, seem to remember, that IPv6 is either un-implemented or
implemented-but-unexercized on Windows.